app/webpacker/components/MapController.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react'
+import SimpleMap from './SimpleMap'
+import { CREATE_TRIP_URL, TRIP_WATCH_URL, CREATE_CHECKINS_URL } from '../constants/api_endpoints'
+import makeNum from '../utils/map_helpers'
+import appendToHost from '../utils/url_helper'
+import axiosPostRequest from '../utils/axios_helper'
+
+class MapController extends Component {
+  state = { checkins: [] }
+
+  componentDidMount() {
+    this.getLocation()
+    this.timer = setInterval(this.updateMap, 5000)
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.timer)
+  }
+
+  getLocation = (name) => {
+    if (!navigator.geolocation) return
+
+    navigator.geolocation.getCurrentPosition((position) => {
+      const data = {
+        lat: position.coords.latitude,
+        lng: position.coords.longitude,
+        name,
+      }
+      this.setState({ startingPoint: data }, this.createTrip)
+    })
+  }
+
+  createTrip = () => {
+    const { startingPoint } = this.state
+    const { lat, lng } = startingPoint
+    axiosPostRequest(CREATE_TRIP_URL, { lat, lng, name: 'DemoTrip' }, this.generateLink)
+  }
+
+  generateLink = (response) => {
+    const { data, included } = response.data
+    this.setState({
+      tripId: data.attributes.id,
+      url: appendToHost(`${TRIP_WATCH_URL}/${data.attributes.uuid}`),
+      checkins: makeNum(included),
+    }, this.updateMap)
+  }
+
+  updateMap = () => {
+    const { startingPoint } = this.state
+    this.setState({ center: startingPoint }, this.updateCurrentLocation)
```

Comment:
`this.setState(({ startingPoint }) => ({ center: startingPoint }), this.updateCurrentLocation)`